tests/tracking/test_client.py

```diff
@@ -500,3 +508,47 @@ def test_get_databricks_runtime_nondb(mock_spark_session):
     runtime = get_databricks_runtime()
     assert runtime is None
     mock_spark_session.conf.get.assert_not_called()
+
+
+def test_client_can_be_serialized_with_pickle(tmpdir):
+    """
+    Verifies that instances of `MlflowClient` can be serialized using pickle, even if the underlying
+    Tracking and Model Registry stores used by the client are not serializable using pickle
+    """
+
+    class MockUnpickleableTrackingStore(SqlAlchemyTrackingStore):
+        pass
+
+    class MockUnpickleableModelRegistryStore(SqlAlchemyModelRegistryStore):
+        pass
+
+    backend_store_path = os.path.join(str(tmpdir), "test.db")
+    artifact_store_path = os.path.join(str(tmpdir), "artifacts")
+
+    mock_tracking_store = MockUnpickleableTrackingStore(
+        "sqlite:///" + backend_store_path, artifact_store_path
+    )
+    mock_model_registry_store = MockUnpickleableModelRegistryStore(
+        "sqlite:///" + backend_store_path
+    )
+
+    # Verify that the mock stores cannot be pickled because they are defined within a function
+    # (i.e. the test function)
+    with pytest.raises(AttributeError):
```

Comment:
```suggestion
    with pytest.raises(AttributeError, match="<locals>.MockUnpickleableTrackingStore'"):
```

Can we specify `match` here to make sure pickle fails to dump the client due to `MockUnpickleableModelRegistryStore`?

Reply:
Done! Thanks!